client/lib/user-settings/index.js

```diff
@@ -204,10 +221,11 @@ UserSettings.prototype.getSetting = function( settingName ) {
 	var setting = null;
 
 	// If we haven't fetched settings, or if the setting doesn't exist return null
-	if ( this.settings && 'undefined' !== typeof this.settings[ settingName ] ) {
-		setting = ( 'undefined' !== typeof this.unsavedSettings[ settingName ] )
-			? this.unsavedSettings[ settingName ]
-			: this.settings[ settingName ];
+	if ( this.settings && 'undefined' !== typeof get( this.settings, settingName ) ) {
```

Comment:
Just a suggestion here, FFTI as it's not necessarily in scope for this PR...

`_.isUndefined` could be useful here, but I think `_.has` would be even more useful.

If IIRC we could squish these checks down like so:

```js
if ( this.settings && 'undefined' !== typeof get( this.settings, settingName ) ) {
// >>
if ( ! has( this.settings, settingName ) ) {

```